sale_commission/wizard/wizard_settle.py

```diff
@@ -85,11 +85,15 @@ def action_settle(self):
                         settlement = settlement_obj.create(
                             {'agent': agent.id,
                              'date_from': sett_from,
-                             'date_to': sett_to})
+                             'date_to': sett_to,
+                             'company_id':
+                                agent_lines[pos].invoice_line.company_id.id})
```

Comment:
Imho this behavior is the correct the users in parent companies should be
for consulting and reporting because if we allow these users can create
settlements the normal users can not access to it by company restrictions.
At most we can check that the settlement don't mix companies.